### PR TITLE
ebpf: use new kernel symbols from libbpfgo

### PR DESCRIPTION
**Pull request by roikol**
use new libbpgo kernel symbols Refresh method so that kernel symbols will be up-to-date throughout all components

## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

```
Author: RoiKol <roi.kol@aquasec.com>
Date:   Tue Jan 3 19:47:02 2023 +0200

    ebpf: use new kernel symbols from libbpfgo
    
    use new libbpgo kernel symbols Refresh method so that
    kernel symbols will be up-to-date throughout all components
```

Fixes: #2526

## Type of change

- [x] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [ ] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

using POC of seq_ops hooks (can be provided).
now we get the correct values for `hooked_seq_ops` event:

`19:19:35:747254  0      insmod           49240   49240   0                hooked_seq_ops       hooked_seq_ops: map[tcp4_seq_ops_show:{hook_tcp4_seq_show main}]`

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [x] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented all functions/methods created explaining what they do.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

**Review by rafaeldtinoco — CHANGES_REQUESTED** [review]
the bump (go.mod and go.sum) should have their own commit (sorry, it has been our standard).

File: pkg/ebpf/tracee.go [hunk 1]
```diff
@@ -384,7 +384,7 @@ func (t *Tracee) Init() error {
 	// Init kernel symbols map
 
 	if initReq.kallsyms {
-		err = t.UpdateKernelSymbols()
+		err = t.NewKernelSymbols()
```
Comment (NDStrahilevitz) [inline pkg/ebpf/tracee.go:387]: I'd like to move to the Lazy symbols in this PR. I've ran our internal e2e on it before so it should be good to go.

**Review by rafaeldtinoco — COMMENTED** [review]
LGTM. I'll let Nadav merge once you change it to the LazyKernelSymbol like he requested (as he has already tested it).

**Comment by roikol** [comment]
> LGTM. I'll let Nadav merge once you change it to the LazyKernelSymbol like he requested (as he has already tested it).

currently waiting for this PR to be merged in libbpfgo - https://github.com/aquasecurity/libbpfgo/pull/283

**Review by NDStrahilevitz — APPROVED** [review]

**Review by rafaeldtinoco — APPROVED** [review]

**Pull request merged**